.github/workflows/build-and-test.yml

```diff
@@ -139,3 +139,167 @@ jobs:
+          Test-Path -Path ".\splunk-support-bundle.ps1"
+          Start-Process ".\otelcol.exe" -WindowStyle Hidden
+          powershell.exe -File "$env:ProgramFiles\Splunk\OpenTelemetry Collector\splunk-support-bundle.ps1" -t \tmp\splunk-support-bundle
+          Test-Path -Path ".\splunk-support-bundle.zip"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\splunk-otel-collector.log"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\splunk-otel-collector.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\td-agent.log"
+          Test-Path -Path "\tmp\splunk-support-bundle\logs\td-agent.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\collector-metrics.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\df.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\free.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\metrics\top.txt"
+          Test-Path -Path "\tmp\splunk-support-bundle\zpages\tracez.html"
+          if ( "${{ matrix.MODE }}" -eq "agent" ) {
+            Test-Path -Path "\tmp\splunk-support-bundle\config\td-agent\td-agent.conf"
+            Test-Path -Path "\tmp\splunk-support-bundle\config\agent_config.yaml"
+          }
+          if ( "${{ matrix.MODE }}" -eq "gateway" ) {
+            Test-Path -Path "\tmp\splunk-support-bundle\config\gateway_config.yaml"
+          }
+
+  windows-choco-validation:
+    name: windows-choco-validation
+    runs-on: windows-latest
+    needs: [windows-msi]
+    strategy:
+      matrix:
+        MODE: [ "agent", "gateway" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Downloading msi build
+        uses: actions/download-artifact@v2
+        with:
+          name: msi-build
+          path: ./dist
+
+      - name: Chocolatey test
+        run: |
+          $ErrorActionPreference = 'Stop'
+          Set-PSDebug -Trace 1
+          $msi_file_name = Resolve-Path .\dist\splunk-otel-collector*.msi | Split-Path -leaf
+          if ($msi_file_name -match '(\d+\.)(\d+\.)+(\d*)') { $version = $matches[0] }
+          write-host "Building choco package..."
+          .\internal\buildscripts\packaging\choco\make.ps1 build_choco -Version $version | Tee-Object -file .\dist\build_logs.log
+          Test-Path -Path ".\dist\splunk-otel-collector.$version.nupkg"
+          write-host "Installing choco package..."
+          choco install splunk-otel-collector -s=".\dist\splunk-otel-collector.$version.nupkg" --params="'/SPLUNK_ACCESS_TOKEN=12345 /MODE:${{ matrix.MODE }}'" -y | Tee-Object -file .\dist\installation_logs.log
+          if (!((Get-CimInstance -ClassName win32_service -Filter "Name = 'splunk-otel-collector'" | Select Name, State).State -Eq "Running")) { throw "Failed to install splunk-otel-collector using chocolatey." } else { write-host "splunk-otel-collector service is running." }
```

Comment:
Please add a step to verify that the `HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment\SPLUNK_CONFIG` registry value matches `${{ matrix.MODE }}_config.yaml`.

Reply:
added this validation